config.proto

```diff
@@ -572,6 +572,25 @@ message Config {
     */
     int32 tx_power = 10;
 
+    /*
+     * This is controlling the actual hardware frequency the radio is transmitting on.
+     * Most users should never need to be exposed to this field/concept.
+     * A channel number between 1 and N (whatever the max is in the current region).
+     * If ZERO then the rule is "use the old channel name hash based
+     * algorithm to derive the channel number")
+     * If using the hash algorithm the channel number will be: hash(channel_name) %
+     * NUM_CHANNELS (Where num channels depends on the regulatory region).
+     * hash a string into an integer - djb2 by Dan Bernstein. -
+     * http://www.cse.yorku.ca/~oz/hash.html
+     * unsigned long hash(char *str) {
+     *   unsigned long hash = 5381; int c;
+     *   while ((c = *str++) != 0)
+     *     hash = ((hash << 5) + hash) + (unsigned char) c;
+     *   return hash;
+     * }
+     */
+    uint32 channel_num = 11;
+
```

Comment:
This comment needs serious revising, what numbers correspond to what frequencies? also, no need for the hash string > int part.

Reply:
I think there are always 8 channels now too

Reply:
here `channel_num` refers to LoRa terminology and is the LoRa channel number (ie. `channel_num = 1` is LoRa Channel 1).

frequency bands have a total number of LoRa channels depending on the bandwidth (BW) used.
--> _total number of channels = (freq_end - freq_start) / BW_

_ex:_ **EU_433/250KHz BW**: freq_end - freq_start = 434,000 - 433,000 = 1000 / 250 = **4 LoRa channels total**.

Channel 1 starts at 433Mhz and ends at 433,250MHz (+1 BW): freq_center = 433,125Mhz
freq_center for Channel 2 = Channel 1 + 1 BW = 433,375Mhz (etc ...)

so frequency center can be calculated from `channel_num`: https://github.com/meshtastic/Meshtastic-device/blob/master/src/mesh/RadioInterface.cpp#L432

Here's a list of total number of channels (**numChannels**) per frequency band ("regions") for 250KHz bandwidth:

<img width="465" alt="Captura de Tela 2022-09-30 às 18 30 04" src="https://user-images.githubusercontent.com/48680741/193361679-9c0b0083-14bf-494e-bb89-4d25f9f68329.png">